chat-plugins/tickets.js

```diff
@@ -0,0 +1,487 @@
+	buf += `</details><br /><details style="margin: 3px"><summary>Something else</summary>If your issue is not handled above, click the button below to ask for a global. Please be ready to explain the situation.<br /><br /><button class="button" name="send" value="/ticket submit Other">Call a global</button></details></div>`;
+	return buf;
+}
+
+function updateTicketList(init, all) {
+	let buf = `>view-ticket-list\n`;
+	if (init) buf += `|init|html\n|title|Ticket List\n`;
+	buf += `|pagehtml|<div class="pad" style="text-align:center"><button class="button" name="send" value="/report list" style="float:left"><i class="fa fa-refresh"></i> Refresh</button><br /><br /><table style="margin-left: auto; margin-right: auto" border="1" cellspacing="0" cellpadding="3"><tbody><tr><th colspan="5">Ticket List</th></tr><tr><th>Status</th><th>Creator</th><th>Ticket Type</th><th>Claimed by</th><th>Action</th></tr>`;
+	let keys = Object.keys(tickets);
+	keys.sort((a, b) => {
+		a = tickets[a];
+		b = tickets[b];
+		if ((a.banned && !b.banned) || (!a.banned && b.banned)) {
+			return (a.banned ? 1 : -1);
+		}
+		if (a.open !== b.open) {
+			return (a.open ? -1 : 1);
+		} else {
+			if (a.open) return a.created - b.created;
+			return b.created - a.created;
+		}
+	});
+	let hasBanHeader = false;
+	for (let i = 0; i < keys.length; i++) {
+		let ticket = tickets[keys[i]];
+		if (ticket.banned) {
+			if (!hasBanHeader) {
+				buf += `<tr><th>Status</th><th>Username</th><th>Banned by</th><th>Expires</th><th>Logs</th></tr>`;
+				hasBanHeader = true;
+			}
+			buf += `<tr><td><span style="color:red"><i class="fa fa-ban"></i> Banned</td><td>${ticket.name}</td><td>${ticket.by}</td><td>${Chat.toDurationString(ticket.expires - Date.now(), {precision: 1})}</td><td><a href="http://logs2.psim.us:8080/help-${ticket.userid}/"><button class="button">View Log</button></a></td></tr>`;
+		} else {
+			buf += `<tr><td>${ticket.open ? `<span style="color:green"><i class="fa fa-circle-o"></i> Open</span>` : `<span style="color:red"><i class="fa fa-check-circle-o"></i> Closed</span>`}</td><td>${ticket.creator}</td><td>${ticket.type}</td><td>${ticket.claimed ? ticket.claimed : (ticket.open ? `<button class="button" name="send" value="/join help-${ticket.userid}">Claim</button>` : `-`)}</td><td>${ticket.open ? `<button class="button" name="send" value="/ticket close ${ticket.userid}">Close</button>` : ``} <a href="http://logs2.psim.us:8080/help-${ticket.userid}/"><button class="button">View Log</button></a></td></tr>`;
+		}
+	}
+	buf += `</tbody></table></div>`;
+	if (all) {
+		Users.users.forEach(u => {
+			if (u.isStaff && u.connected) u.send(buf);
```

Comment:
... What if a staff is not dealing with tickets? They are basically forced to have the page up...?

Reply:
Client drops the message since it dosen't have that room up.

Reply:
As I mentioned 44 minutes ago, manual refresh is a better model overall and also saves bandwidth.